src/online2bin/online2-tcp-nnet3-decode-faster.cc

```diff
@@ -85,6 +86,27 @@ std::string LatticeToString(const Lattice &lat, const fst::SymbolTable &word_sym
   return msg.str();
 }
 
+std::string GetTimeString(int32 t_beg, int32 t_end, int32 subsamp) {
+  char buffer[100];
+  double t_beg2 = t_beg * subsamp / 100.;
```

Comment:
I don't think we should just assume that the frame shift is 0.01.  There must be a way to infer the actual frame shift from the feature_opts.  I'm not sure why it has to be %5.2f instead of %.2f?  Please document this with an example of the kind of string it returns.  And I'm not 100% convinced that the {, --> and } is helpful; I feel like just the plain numbers might be easier for client code to parse.  Also it's not clear to me that the end-time of the last word, as opposed to the length of time that was decoded, is the most useful thing.  I am starting to wonder whether it might make more sense just to let people make similar changes depending on their use-case.

Reply:
Okay, but how can I get the 'length of time that was decoded'? Is there an easy way to measure the 'time-lenght' of the Lattice?

Reply:
Got it, getting alignment and measuring its length...